### PR TITLE
refactor checkbox handling

### PR DESCRIPTION
**Pull request by thomaskneisel**
check and use selected checkbox before placing the belts

This fixes the issue https://github.com/matjojo/BeltBrush/issues/1
AND some refactoring :relaxed: ... there is more to refactoring 

* its possible to have just one place method/function and inject the direcretion
* use more `local` you don't need all this public global vars ... less is more ;)

/cc @matjojo

### CODE REVIEW

**Comment by matjojo** [comment]
For the fix for issue #1 take a look at that thread, for the local variables;

Does that make a difference in CPU-time? as I understood it, every mod has its own dataspace in factorio, so it wouldn't really matter what you use, I might be wrong though.

For the place method:

That is indeed possible, but this seemed a bit easier at the time of writing, I know it could be done better, but for readability I choose this.

File: control.lua [hunk 1]
```diff
@@ -146,56 +123,34 @@ script.on_event(defines.events.on_built_entity, function(event)
...
-			BB_glob_amount = {}
-			BB_glob_amount[i] = 0
-			game.players[i].gui.top.BB_flow_main.add{type = "label", name = "BB_label_2", caption = "2"}
-			game.players[i].gui.top.BB_flow_main.add{type = "checkbox", name = "BB_check_2", state = false}
-			game.players[i].gui.top.BB_flow_main.add{type = "label", name = "BB_label_3", caption = "3"}
-			game.players[i].gui.top.BB_flow_main.add{type = "checkbox", name = "BB_check_3", state = false}
-			game.players[i].gui.top.BB_flow_main.add{type = "label", name = "BB_label_4", caption = "4"}
-			game.players[i].gui.top.BB_flow_main.add{type = "checkbox", name = "BB_check_4", state = false}
+	
+	for i,player in pairs(game.players) do
```
Comment (matjojo) [inline control.lua]: when you do Ipairs, Lua only goes through the numbers in an array or table, that makes it take a bit less CPU-time/be some more efficient.

File: control.lua [hunk 2]
```diff
@@ -146,56 +123,34 @@ script.on_event(defines.events.on_built_entity, function(event)
...
-	test = function()
-
-		for i in ipairs(game.players) do
-
-			game.players[i].print("amount = " .. tostring(game.players[i].get_inventory(defines.inventory.player_main).get_item_count("basic-transport-belt")))
-
-		end
-
+	local player = game.players[event.player_index]
+	for checkbox = 1, 4, 1 do
```
Comment (matjojo) [inline control.lua]: wouldn't this set every checkbox to false?

File: control.lua (hunk 1 above)
Comment (thomaskneisel) [inline control.lua]: yes, but you want the player object anyway :)

File: control.lua (hunk 2 above)
Comment (thomaskneisel) [inline control.lua]: yes, its common for those sets, set all to false, and the "ONE" to true. Easier to read and understand.

**Comment by thomaskneisel** [comment]
... sry the programmer in me just rewrite allmost everthing in the code ... ask if you have qustions :D

**Comment by matjojo** [comment]
Hi, I have put some thought in you being willing to help and how I feel about that. First of all I feel like I need to thank you, you probably put a bit of time in the work you did, and I really am grateful for that.

But. I do this for fun, and to learn myself some programming. Sadly I have been put off off this mod jus for the reason that you where helping. It started to feel too much of a MUST do instead of a CAN do. And I really disliked the feeling that that gave. After giving it a lot of though (about two weeks) I came to the conclusion that I will keep the code on Github as I do want to give people a way to look it through easily. But, I am not going to make use of the help of you, or anyone else for that order.

I again want to thank you for being willing to help.

~Matjojo

**Pull request closed without merging**